numpy/distutils/fcompiler/gnu.py

```diff
@@ -57,7 +57,7 @@ def gnu_version_match(self, version_string):
                     return ('gfortran', m.group(1))
         else:
             # Output probably from --version, try harder:
-            m = re.search(r'GNU Fortran\s+95.*?([0-9-.]+)', version_string)
+            m = re.search(r'95.*?([0-9-.]+)', version_string)
```

Comment:
The increases generality is a bit bothersome. Couldn't this be done with an `if` to cover both cases more precisely.